packages/babel-parser/src/parser/expression.js

```diff
@@ -1942,8 +1943,8 @@ export default class ExpressionParser extends LValParser {
     refExpressionErrors?: ?ExpressionErrors,
   ): N.ObjectMember | N.SpreadElement | N.RestElement {
     let decorators = [];
-    if (this.match(tt.at)) {
-      if (this.hasPlugin("decorators")) {
+    if (this.match(tt.at) || this.match(tt.atInit)) {
+      if (this.hasPlugin("decorators") || this.match(tt.atInit)) {
```

Comment:
`@init:` can only be used then the `decorators` plugin is enabled, so it's enough to use `if (this.hasPlugin("decorators")) {`.